controllers/apps/operations/datascript.go

```diff
@@ -0,0 +1,283 @@
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	appsv1alpha1 "github.com/apecloud/kubeblocks/apis/apps/v1alpha1"
+	"github.com/apecloud/kubeblocks/internal/constant"
+	componetutil "github.com/apecloud/kubeblocks/internal/controller/component"
+	intctrlutil "github.com/apecloud/kubeblocks/internal/controllerutil"
+	sqlchannelutil "github.com/apecloud/kubeblocks/internal/sqlchannel/util"
+)
+
+var _ OpsHandler = DataScriptOpsHandler{}
+var _ error = &FastFaileError{}
+
+// DataScriptOpsHandler handles DataScript operation, it is more like a one-time command operation.
+type DataScriptOpsHandler struct {
+}
+
+// FastFaileError is a error type that will not retry the operation.
+type FastFaileError struct {
+	message string
+}
+
+func (e *FastFaileError) Error() string {
+	return fmt.Sprintf("fail with message: %s", e.message)
+}
+
+func init() {
+	// ToClusterPhase is not defined, because 'datascript' does not affect the cluster status.
+	dataScriptOpsHander := DataScriptOpsHandler{}
+	dataScriptBehavior := OpsBehaviour{
+		FromClusterPhases:          []appsv1alpha1.ClusterPhase{appsv1alpha1.RunningClusterPhase},
+		MaintainClusterPhaseBySelf: false,
+		OpsHandler:                 dataScriptOpsHander,
+	}
+	opsMgr := GetOpsManager()
+	opsMgr.RegisterOps(appsv1alpha1.DataScriptType, dataScriptBehavior)
+}
+
+// Action implements OpsHandler.Action
+// It will create a job to execute the script. It will fail fast if the script is not valid, or the target pod is not found.
+func (o DataScriptOpsHandler) Action(reqCtx intctrlutil.RequestCtx, cli client.Client, opsResource *OpsResource) error {
+	opsRequest := opsResource.OpsRequest
+	cluster := opsResource.Cluster
+	spec := opsRequest.Spec.ScriptSpec
+
+	// get component
+	component := cluster.Spec.GetComponentByName(spec.ComponentName)
+	if component == nil {
+		// we have checked component exists in validation, so this should not happen
+		return &FastFaileError{message: fmt.Sprintf("component %s not found in cluster %s", spec.ComponentName, cluster.Name)}
+	}
+
+	clusterDef := &appsv1alpha1.ClusterDefinition{}
+	if err := cli.Get(reqCtx.Ctx, types.NamespacedName{Name: cluster.Spec.ClusterDefRef}, clusterDef); err != nil {
+		if apierrors.IsNotFound(err) {
+			// fail fast if cluster def does not exists
+			return &FastFaileError{message: err.Error()}
+		}
+		return err
+	}
+	// get componentDef
+	componentDef := clusterDef.GetComponentDefByName(component.ComponentDefRef)
+	if componentDef == nil {
+		return &FastFaileError{message: fmt.Sprintf("componentDef %s not found in clusterDef %s", component.ComponentDefRef, clusterDef.Name)}
+	}
+
+	// parse scripts
+	script, err := getScriptContent(reqCtx, cli, spec)
+	if err != nil {
+		return &FastFaileError{message: err.Error()}
+	}
+
+	scriptRequest := sqlchannelutil.SQLChannelRequest{
+		Operation: "exec",
+		Metadata: map[string]interface{}{
+			"sql": strings.Join(script, " "),
+		},
+	}
```

Comment:
how about the timeout of SQLChannelRequest?

Reply:
It is 1s by default. The timeout option will be exposed to user when all such execution requests support the setting later.